### ISSUE TITLE
Switch to new rust-bindgen crate

### ISSUE THREAD

**Issue by nlfiedler** [nlfiedler/magick-rust#22]
The old rust-bindgen crate (https://github.com/Yamakaky/rust-bindgen) has been deprecated, and Mozilla has taken over development. The new crate is at https://github.com/servo/rust-bindgen and a tutorial of sorts is available here: http://fitzgeraldnick.com/2016/12/14/using-libbindgen-in-build-rs.html

Note that we need to set environment variables in order to build the ImageMagick bindings, so using the library may not be feasible. But at least we can switch to the new crate.

This may help with issue #16.

**Comment by nlfiedler** [comment]
The tutorial builds and tests successfully on MacOS, while on FreeBSD 11 it generates numerous warnings about a `non-foreign-function-safe` struct member, but otherwise is successful. Using this crate may indeed fix issue #16.

**Comment by nlfiedler** [comment]
The new version of bindgen is different and not all of the old options are available any more. As such, building bindings for MagickWand is very difficult now.

**Comment by bbigras** [comment]
Any progress on this?

**Comment by nlfiedler** [comment]
Spent an hour looking at it again. It does not look like the latest version offers the missing features, such as setting environment variables. But before that, need to determine why it refuses to see the ImageMagick headers. Both macOS and FreeBSD have the same problem, cannot find the `MagickWand.h` file, despite passing the CFLAGS and LDFLAGS to bindgen.

**Comment by gadomski** [comment]
@nlfiedler I've got the beginnings of a conversion to the bindgen crate [here](https://github.com/gadomski/magick-rust/tree/issues/22-bindgen). I'm finding the headers ok, but there's other problems with the generated bindings, including matching up exception types to the wrapper. I spent a little bit on it yesterday and might dig some more today, but I thought I'd throw that up just in case that helps things on your end.

I'm working on OSX ATM.
